honeybee/room.py

```diff
@@ -0,0 +1,605 @@
+        """A number between 0 and 360 for the average orientation of exposed walls.
+
+        0 = North, 90 = East, 180 = South, 270 = West.  Wil be None if the zone has
+        no exterior walls. Resulting value is weighted by the area of each of the
+        wall faces.
+
+        Args:
+            north_vector: A ladybug_geometry Vector2D for the north direction.
+                Default is the Y-axis (0, 1).
+        """
+        orientations = 0
+        areas = 0
+        for face in self._faces:
+            if isinstance(face.type, Wall) and \
+                    isinstance(face.boundary_condition, Outdoors):
+                orientations += face.horizontal_orientation(north_vector)
+                areas += face.area
+        return orientations / areas if areas != 0 else None
+
+    def clear_shades(self):
+        """Remove all indoor and outdoor shades assigned to this room."""
+        self.clear_indoor_shades()
+        self.clear_outdoor_shades()
+
+    def clear_indoor_shades(self):
+        """Remove all indoor shades assigned to this room."""
+        for shade in self._indoor_shades:
+            shade._parent = None
+        self._indoor_shades = []
+
+    def clear_outdoor_shades(self):
+        """Remove all outdoor shades assigned to this room."""
+        for shade in self._outdoor_shades:
+            shade._parent = None
+        self._outdoor_shades = []
+
+    def add_indoor_shade(self, shade):
+        """Add a Shade object to be added to the indoor of this room.
+
+        Indoor Shade objects can be used to represent furniture, the interior
+        part of light shelves, etc.
+        For representing finely detailed objects like blinds or roller shades,
+        it may be more apprpriate to model them as materials assigned to
+        Aperture properties (like Radiance or Energy materials).
+
+        Args:
+            shade: A Shade object to add to the indoors of this room.
+        """
+        assert isinstance(shade, Shade), \
+            'Expected Shade for indoor_shade. Got {}.'.format(type(shade))
+        shade._parent = self
+        self._indoor_shades.append(shade)
+
+    def add_outdoor_shade(self, shade):
+        """Add an Shade object to the outdoor of this room.
+
+        Exterior Shade objects can be used to represent overhangs, fins, etc.
+        For representing larger shade objects like trees or other buildings,
+        it may be more appropriate to add them to the Model as standalone shades
+        without a specific parent Room since they can shade multiple Rooms at once.
+
+        Args:
+            shade: A shade face to add to the outdoors of this room.
+        """
+        assert isinstance(shade, Shade), \
+            'Expected Shade for outdoor_shade. Got {}.'.format(type(shade))
+        shade._parent = self
+        self._outdoor_shades.append(shade)
+
+    def generate_mesh_grid(self, x_dim, y_dim=None, offset=1.0):
```

Comment:
I know the results will be a mesh but not sure if we need to include it in the name of the method. You can just call it `generate_grid`.

Reply:
I am fine with that if you think `generate_grid` is clear enough.  I will make the change and I'll make sure the docstring elaborates that the output is a mesh.